src/index.ts

```diff
@@ -277,12 +263,234 @@ export class ConsentControl {
      * 
      * Apply the theme and change banner and preferences dialog's color 
      * 
-     * TODO
-     * 
      * @param {string} themeName theme that will be applied
      */
     public applyTheme(themeName: string): void {
-        ;
+        if (!this.themes[themeName]) {
+            throw new Error("Theme not found error");
+        }
+
+        let theme: ITheme = <ITheme> this.themes[themeName];
+
+        // overwrite banner styles
+        let bannerStyle = document.createElement('style');
+        bannerStyle.type = 'text/css';
+        bannerStyle.innerHTML = `.${ styles.bannerBody } {
+            background-color: ${ theme["banner-background-color"] } !important;
+        }`;
+
+        document.getElementsByTagName('head')[0].appendChild(bannerStyle);
+
+        // overwrite text color styles
+        let textColorStyle = document.createElement('style');
+        textColorStyle.type = 'text/css';
+        textColorStyle.innerHTML = `.${ styles.textColorTheme } {
+            color: ${ theme["text-color"] } !important;
+        }`;
+
+        document.getElementsByTagName('head')[0].appendChild(textColorStyle);
+        
+        // overwrite hyper link styles
+        let hyperLinkStyle = document.createElement('style');
+        hyperLinkStyle.type = 'text/css';
+        hyperLinkStyle.innerHTML = `.${ styles.hyperLinkTheme } a {
+            color: ${ theme["hyperlink-font-color"] } !important;
+        }`;
+
+        document.getElementsByTagName('head')[0].appendChild(hyperLinkStyle);
+        
+        this.setDialogStyle(theme);
```

Comment:
Let's extract all these methods to a separate module. I suggest we create a separate folder `themes` and put everything related to themes (including `theme-styles.ts` and this set...Style() ) to that folder.